### PR TITLE
various add-ons: More example alerts

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOGS > Added change notes.
- Scan rules > Added example alert functionality.
- UnitTests > Added tests for example alerts.

## Related Issues
- zaproxy/zaproxy#6119
- zaproxy/zaproxy#7100
- zaproxy/zaproxy#8262

## Checklist
- [NA] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/ShellShockScanRule.java [hunk 1]
```diff
@@ -49,11 +50,18 @@ public class ShellShockScanRule extends AbstractAppParamPlugin {
     /** the logger object */
     private static final Logger LOGGER = LogManager.getLogger(ShellShockScanRule.class);
 
-    private final String attackHeader = HttpFieldsNames.X_POWERED_BY;
+    private static final String ATTACK_HEADER = HttpFieldsNames.X_POWERED_BY;
 
-    // Use a standard HTTP response header, to make sure the header is not dropped by load
-    // balancers, proxies, etc
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/ShellShockScanRule.java:55]: This was probably for the above field (evidence is not a response header).

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/ShellShockScanRule.java [hunk 2]
```diff
@@ -49,11 +50,18 @@ public class ShellShockScanRule extends AbstractAppParamPlugin {
...
+    /**
+     * Use a standard HTTP response header, to make sure the header is not dropped by load
+     * balancers, proxies, etc
+     */
+    private static final String EVIDENCE = "ShellShock-Vulnerable";
+
+    private static final String ATTACK_1 =
+            "() { :;}; echo '" + ATTACK_HEADER + ": " + EVIDENCE + "'";
+    private static final String ATTACK_2 = "() { :;}; /bin/sleep "; // Will have sleep value
+    // concatenated on the end
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/ShellShockScanRule.java:64]: Add comment before the line, fits better.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/ShellShockScanRule.java [hunk 3]
```diff
@@ -196,4 +207,11 @@ public int getWascId() {
     public Map<String, String> getAlertTags() {
         return ALERT_TAGS;
     }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return List.of(
+                buildAlert("profile", ATTACK_1).build(),
+                buildTimingAlert("profile", ATTACK_2 + "5", TimeUnit.SECONDS.toMillis(9L)).build());
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/ShellShockScanRule.java:215]: I'd have expected the same sleep value (or at least a closer value).

File: addOns/pscanrules/CHANGELOG.md [hunk 4]
```diff
@@ -13,7 +13,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
     - User Controllable JavaScript Event (XSS)
     - Cookie without SameSite Attribute
     - X-Debug-Token Information Leak
-- The Cookie without SameSite Attribute scan rule now has alert references (Issue 7100).
+    - Retrieved from Cache
+- The following scan rules now have alert references (Issue 7100):
+    - Cookie without SameSite Attribute 
+    - Retrieved from Cache (invalid raw text was also trimmed from the Alert's references (Issue 8262))
```
Comment (thc202) [inline addOns/pscanrules/CHANGELOG.md:19]: - Not sure about calling it invalid raw text, we didn't have any rules about what the references could have or not.
- Should be used singular, only one reference was changed.

Comment (kingthorin) [reply addOns/pscanrules/CHANGELOG.md:19]: Okay, I had said "invalid" because it was making the new reference testing fail. I've tweaked it.

**Comment by kingthorin** [comment]
Review addressed.

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 5]
```diff
@@ -9,7 +9,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
     - Exponential Entity Expansion (Billion Laughs Attack)
     - Relative Path Confusion
     - HTTPS Content Available via HTTP
-- The HTTPS Content Available via HTTP rule now includes example alert functionality for documentation generation purposes (Issue 6119).
+    - Remote Code Execution - Shell Shock
+- The following scan rules now includes example alert functionality for documentation generation purposes (Issue 6119):
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/CHANGELOG.md:13]: include

**Review by thc202 — APPROVED** [review]

**Review by psiinon — APPROVED** [review]

**Pull request merged**